frontend/src/lib/components/CommandBar/searchBarLogic.ts

```diff
@@ -116,35 +206,135 @@ export const searchBarLogic = kea<searchBarLogicType>([
+                s.searchQuery,
+            ],
+            (
+                searchResponse,
+                personsResponse,
+                group0Response,
+                group1Response,
+                group2Response,
+                group3Response,
+                group4Response,
+                query
+            ) => {
+                if (
+                    !searchResponse &&
+                    !personsResponse &&
+                    !group0Response &&
+                    !group1Response &&
+                    !group2Response &&
+                    !group3Response &&
+                    !group4Response
+                ) {
                     return null
                 }
 
                 return [
                     ...(searchResponse ? searchResponse.results : []),
                     ...(personsResponse ? rankPersons(personsResponse.results, query) : []),
+                    ...(group0Response ? rankGroups(group0Response.results, query) : []),
+                    ...(group1Response ? rankGroups(group1Response.results, query) : []),
+                    ...(group2Response ? rankGroups(group2Response.results, query) : []),
+                    ...(group3Response ? rankGroups(group3Response.results, query) : []),
+                    ...(group4Response ? rankGroups(group4Response.results, query) : []),
                 ].sort((a, b) => (a.rank && b.rank ? a.rank - b.rank : 1))
             },
         ],
         combinedSearchLoading: [
-            (s) => [s.rawSearchResponseLoading, s.rawPersonsResponseLoading],
-            (searchLoading, personsLoading) => searchLoading && personsLoading,
+            (s) => [
+                s.rawSearchResponseLoading,
+                s.rawPersonsResponseLoading,
+                s.rawGroup0ResponseLoading,
+                s.rawGroup1ResponseLoading,
+                s.rawGroup2ResponseLoading,
+                s.rawGroup3ResponseLoading,
+                s.rawGroup4ResponseLoading,
+            ],
+            (
+                searchLoading,
+                personsLoading,
+                group0Loading,
+                group1Loading,
+                group2Loading,
+                group3Loading,
+                group4Loading
+            ) =>
+                searchLoading &&
+                personsLoading &&
+                group0Loading &&
+                group1Loading &&
+                group2Loading &&
+                group3Loading &&
+                group4Loading,
+        ],
+        tabs: [
+            (s) => [s.groupTypes],
+            (groupTypes): Tab[] => {
+                return [
+                    Tab.All,
+                    Tab.EventDefinition,
+                    Tab.Action,
+                    Tab.Person,
+                    Tab.Cohort,
+                    ...Array.from(groupTypes.values()).map(
+                        ({ group_type_index }) => `group_${group_type_index}` as Tab
+                    ),
```

Comment:
Could possibly be a selector of its own

Reply:
I'll likely group the tabs in a follow up, leaving this refactor for then.